ansible_collections/arista/avd/roles/eos_designs/python_modules/connected_endpoints/utils.py

```diff
@@ -0,0 +1,287 @@
+            if "parent_profile" in port_profile:
+                parent_profile = get_item(self._port_profiles, "profile", port_profile["parent_profile"], default={})
+                port_profile = merge(parent_profile, port_profile, list_merge="replace", destructive_merge=False)
+                port_profile.pop("parent_profile")
+            merged_port_profiles.append(port_profile)
+
+        return merged_port_profiles
+
+    @cached_property
+    def _connected_endpoints_keys(self) -> list:
+        """
+        Return connected_endpoints_keys converted to list
+        """
+        return convert_dicts(get(self._hostvars, "connected_endpoints_keys", default=[]), "key")
+
+    @cached_property
+    def _hostname(self) -> str:
+        return get(self._hostvars, "switch.hostname", required=True)
+
+    @cached_property
+    def _link_tracking_groups(self) -> list | None:
+        return get(self._hostvars, "switch.link_tracking_groups")
+
+    @cached_property
+    def _mlag(self) -> bool:
+        return get(self._hostvars, "switch.mlag") is True
+
+    @cached_property
+    def _enable_trunk_groups(self) -> bool:
+        return get(self._hostvars, "switch.enable_trunk_groups") is True
+
+    @cached_property
+    def _platform_settings_feature_support_interface_storm_control(self) -> bool:
+        return get(self._hostvars, "switch.platform_settings.feature_support.interface_storm_control", default=True) is True
+
+    @cached_property
+    def _evpn_short_esi_prefix(self) -> str:
+        return get(self._hostvars, "evpn_short_esi_prefix", required=True)
+
+    def _get_short_esi(self, adapter: dict, channel_group_id: int, short_esi: str = None, hash_extra_value: str = "") -> str | None:
+        """
+        Return short_esi for one adapter
+        """
+        if len(set(adapter["switches"])) < 2:
+            # Only configure ESI for multi-homing.
+            return None
+
+        # short_esi is only set when called from sub-interface port-channels.
+        if short_esi is None:
+            # Setting short_esi under port_channel will be deprecated in AVD4.0
+            port_channel_short_esi = get(adapter, "port_channel.short_esi")
+            if (short_esi := get(adapter, "ethernet_segment.short_esi", default=port_channel_short_esi)) is None:
+                return None
+
+        endpoint_ports: list = default(
+            adapter.get("endpoint_ports"),
+            adapter.get("server_ports"),
+        )
+        short_esi = str(short_esi)
+        if short_esi.lower() == "auto":
+            esi_hash_switches = "".join(adapter["switches"][:2])
+            esi_hash_switch_ports = "".join(adapter["switch_ports"][:2])
+            esi_hash_endpoint_ports = "".join(endpoint_ports[:2])
+            esi_hash = sha256(
+                f"{hash_extra_value}{esi_hash_switches}{esi_hash_switch_ports}{esi_hash_endpoint_ports}{channel_group_id}".encode("UTF-8")
+            ).hexdigest()
```

Comment:
https://docs.python.org/3/library/hashlib.html
```suggestion
        if short_esi.lower() == "auto":
            esi_hash = sha256()
            esi_hash.update("".join(adapter["switches"][:2]).encode("UTF-8"))
            esi_hash.update("".join(adapter["switch_ports"][:2]).encode("UTF-8"))
            esi_hash.update("".join(endpoint_ports[:2]).encode("UTF-8"))
            esi_hash = esi_hash.hexdigest()
```

Reply:
done a variation